explorer/src/test/java/com/scalar/client/tool/explorer/ExplorerTest.java

```diff
@@ -175,48 +140,55 @@ public void scan_ThrowExceptionOnlyOneTime_ShouldReturnSuccess() {
   public void scan_FailedResponseWhenRegisteringCertificate_ShouldThrowExplorerException() {
     // Arrange
     when(clientService.executeContract(eq(HOLDER_ID + "SCAN"), any(JsonObject.class)))
-        .thenReturn(contractExecutionResponseCertificateNotFound);
-    when(clientService.registerCertificate()).thenReturn(ledgerServiceResponseCertificateNotFound);
+        .thenThrow(clientExceptionCertificateNotFound);
+    doAnswer((i) -> { return null;} ).when(clientService).registerCertificate();
 
     // Act-assert
     assertThatThrownBy(() -> explorer.scan(ASSET_ID, Json.createObjectBuilder().build()))
         .isInstanceOf(ExplorerException.class)
-        .hasMessage("403 certificate not found");
+        .hasMessage("403 Certificate not found");
   }
 
   @Test
-  public void scan_CertificateNotFoundTwice_ShouldThrowExplorerException() {
+  public void get_ContractNotFoundTwice_ShouldThrowExplorerException() {
+    // Arrange
+    when(clientService.executeContract(eq(HOLDER_ID + "GET"), any(JsonObject.class)))
+            .thenThrow(clientExceptionContractNotFound);
+
+    // Act-assert
+    assertThatThrownBy(() -> explorer.get(ASSET_ID))
+            .isInstanceOf(ExplorerException.class)
+            .hasMessage("404 contract not found");
+  }
+  @Test
+  public void scan_ContractNotFoundTwice_ShouldThrowsExplorerException() {
     // Arrange
     when(clientService.executeContract(eq(HOLDER_ID + "SCAN"), any(JsonObject.class)))
-        .thenReturn(
-            contractExecutionResponseCertificateNotFound,
-            contractExecutionResponseCertificateNotFound);
+            .thenThrow(clientExceptionContractNotFound);
 
     // Act-assert
     assertThatThrownBy(() -> explorer.scan(ASSET_ID, Json.createObjectBuilder().build()))
-        .isInstanceOf(ExplorerException.class)
-        .hasMessage("403 certificate not found");
+            .isInstanceOf(ExplorerException.class)
+            .hasMessage("404 contract not found");
   }
-
   @Test
-  public void scan_ContractNotFoundTwice_ShouldThrowsExplorerException() {
+  public void scan_CertificateNotFoundTwice_ShouldThrowExplorerException() {
```

Comment:
Ditto.

Reply:
Thank you for review comment.
Code modified with the above change.
Sorry, I forgot to rename.